IDEAS/Buildings/Components/Shading/BuildingShade.mo

```diff
@@ -24,37 +32,60 @@ model BuildingShade
   // Computation assumes that window base is at ground level.
   // Viewing angle computed from center of glazing.
 protected
+  final parameter Boolean haveOverhang = LOv > Modelica.Constants.small 
+    "To avoid divisions by zero and obsolete computations";
   parameter Modelica.Units.SI.Angle vieAngObj=atan((hWin/2 + dh)/L)
     "Viewing angle of opposite object";
+  parameter Modelica.Units.SI.Angle vieAngOv=if haveOverhang then Modelica.Constants.pi/2 - atan((hWin/2 + dhOv)/LOv) else 0
+    "Viewing angle of overhang";
   final parameter Modelica.Units.SI.Angle rot=0
     "Rotation angle of opposite building. Zero when parallel, positive when rotated clockwise"
     annotation (Evaluate=true);
-  final parameter Real coeff = 1-fraSha "More efficient implementation";
-  final parameter Real hWinInv = 1/hWin "More efficient implementation";
+  final parameter Real coeff = 1-fraSha 
+    "More efficient implementation";
+  final parameter Real coeffOv = 1-fraShaOv 
+    "More efficient implementation";
+  final parameter Real hWinInv = 1/hWin 
+    "More efficient implementation";
   Real tanZen = tan(min(angZen, Modelica.Constants.pi/2.01));
-  Modelica.Units.SI.Length L1
+  Modelica.Units.SI.Length L1 = max(0, L/cos(verAzi))
     "Horizontal distance to object when following vertical plane through sun ray";
-  Modelica.Units.SI.Length L2
-    "Distance to object, taking into account sun position";
-  Modelica.Units.SI.Angle alt=(Modelica.Constants.pi/2) - angZen;
-  Modelica.Units.SI.Angle verAzi
+  Modelica.Units.SI.Length L2 =  L1*tan(alt)
+    "Maximum height of object before it casts some shade";
```

Comment:
"distant object" or "obstruction" (clarify now that the model has two different implementations)